src/server.js

```diff
@@ -0,0 +1,50 @@
+const Bcrypt = require('bcrypt');
+const Hapi = require('hapi');
+const getUserFromDatabase = require('../src/getUsersFromDatabase.js');
+
+const validate = async (request, username, password, h) => {
+
+    const user = await getUserFromDatabase.getUserFromDatabase(username);
+
+    if (!user) {
+        console.log(user);
+        return { credentials: null, isValid: false };
+    }
+
+    const isValid = await Bcrypt.compareSync(password, user.password);
+    const credentials = {
+        username : user.username,
+        firstName: user.firstName,
+        lastName: user.lastName
+    };
+
+    return { isValid, credentials };
+};
+
+const main = async () => {
+
+    const server = Hapi.server({ port: 8080 });
```

Comment:
we can move some of the configuration into config using config library. port can go into config.